packages/web-components/src/radio-group/radio-group.spec.ts

```diff
@@ -233,6 +233,65 @@ test.describe('RadioGroup', () => {
     await expect(radios.nth(2)).toHaveJSProperty('checked', false);
   });
 
+  test('radio should remain checked after it is set to disabled and uncheck when a new radio is checked', async ({
+    page,
+  }) => {
+    const element = page.locator('fluent-radio-group');
+    const radios = element.locator('fluent-radio');
+
+    await page.setContent(/* html */ `
+      <fluent-radio-group>
+        <fluent-radio id="radio-1" name="radio" value="foo"></fluent-radio>
+        <fluent-radio id="radio-2" name="radio" value="bar"></fluent-radio>
+        <fluent-radio id="radio-3" name="radio" value="baz"></fluent-radio>
+      </fluent-radio-group>
+    `);
+
+    await radios.nth(0).evaluate((node: Radio) => {
+      node.checked = true;
+    });
+
+    await radios.nth(0).evaluate((node: Radio) => {
+      node.disabled = true;
+    });
+
+    await expect(radios.nth(0)).toHaveJSProperty('checked', true);
+
+    await radios.nth(1).click();
+
+    await expect(radios.nth(1)).toBeFocused();
```

Comment:
Should the `expect` assertions happen here too? Maybe there should be a few more tests:

- should uncheck all disabled radios when an enabled and unchecked radio is clicked
- should uncheck all disabled radios when the space key is pressed on a focused, enabled, and unchecked radio

Reply:
I would say that if we still needed to run the unchecking of disabled items code inside the keydown handler but since is isolated to the `changeHandler` the test coverage is sufficient.